### ISSUE TITLE
make all under dockerfile

### ISSUE THREAD

**Issue by nickkimer** [ls1xt/deepdecoder#4]
Thanks for providing the code to your paper! Very insightful and helpful to see the code behind the results. I am trying to replicate and have run into this issue. 

After running make all to create docker base-image, I run into this problem -- are there deprecated dependencies or is this a result of my environment?

`
E: Unable to find a source package for libboost-all-dev
Picking 'opencv' as source package instead of 'libopencv-dev'
Picking 'boost-defaults' as source package instead of 'libboost-all-dev'
The command '/bin/sh -c apt-get update && apt-get build-dep -y --assume-yes --no-install-recommends         libopencv-dev         libboost-all-dev &&     rm -rf /var/lib/apt/lists' returned a non-zero code: 100
Makefile:8: recipe for target 'base-image' failed
make: *** [base-image] Error 100
`

**Comment by ls1xt** [comment]
Hi Nick,

have you managed to solve this problem? If so, how did you do it?
This error should not occur. 

Cheers, Leon

**Comment by nickkimer** [comment]
Leon, 

Thanks for checking up on the issue. I looked around and came across the solution here (running on EC2 ubuntu 16.04) https://github.com/silarsis/docker-proxy/issues/21. I put the solution at the top of your dockerfile as:

`RUN sed -i 's/^# deb-src/deb-src/' /etc/apt/sources.list`

and it worked. 

I have fully built your docker image according to your instructions on the readme but I had to make a few changes to get it to build completely. I am currently making a few suggested changes and will add it in as a pull request for your review.
